packages/gatsby-theme-store/src/gatsby-node.ts

```diff
@@ -178,17 +180,30 @@ export const createPages = async ({
   })
 }
 
-export const onCreateWebpackConfig = ({
-  actions: { setWebpackConfig },
-}: CreateWebpackConfigArgs) => {
+export const onCreateWebpackConfig = (
+  { actions: { setWebpackConfig }, stage }: CreateWebpackConfigArgs,
+  { profiling = false }: Options
+) => {
+  const profilingConfig =
+    stage === 'build-javascript' && profiling === true
```

Comment:
this comparison is needed? once profiling will always be boolean

```suggestion
    stage === 'build-javascript' && profiling
```

Reply:
It's always better to explicitly compare values so V8 can better optimize this